src/Network/NWParameters.cs

```diff
@@ -177,6 +177,29 @@ public unsafe static NWParameters CreateUdp (Action<NWProtocolOptions> configure
 			return new NWParameters (ptr, owns: true);
 		}
 
+		[NoWatch, NoTV, NoiOS, Mac (10,15)]
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern IntPtr nw_parameters_create_custom_ip (byte custom_ip_protocol_number, void *configure_ip);
+
+		[NoWatch, NoTV, NoiOS, Mac (10,15)]
+		public unsafe static NWParameters CreateCustomIP (byte protocolNumber, Action<NWProtocolOptions> configureCustomIP = null)
```

Comment:
are there performance differences between declaring the function as `unsafe` vs encapsulating the code (i think most of it here) in an `unsafe` block?

Reply:
Marking the whole method as unsafe does prevent some optimizations (no idea on a specific list) being applied to the whole method as just to the respective block of code - as unsafe code can be used in the whole method now.